ZipArchive.cs

```diff
@@ -691,12 +691,36 @@ public ZipEntry ReadEntry (string entryName, bool caseSensitive = false)
 			return ReadEntry ((ulong)LookupEntry (entryName, caseSensitive));
 		}
 
-		public ZipEntry ReadEntry (ulong index)
+		/// <summary>
+		/// Read a zip entry, given an index.
+		/// 
+		/// By default, if the entry is deleted then an exception is thrown (the error will be ErrorCode.Deleted or
+		/// ErrorCode.Inval, depending on whether the deleted entry previously existed in the zip or was newly
+		/// added - that's just how libzip handles that). If returnNullIfDeleted is true, then null is returned
+		/// for deleted entries and an exception is just thrown for other errors.
+		/// </summary>
+		/// <param name="index">index to read</param>
+		/// <param name="returnNullIfDeleted">whether to return null or throw an exception for deleted entries</param>
+		/// <returns></returns>
+		public ZipEntry ReadEntry (ulong index, bool returnNullIfDeleted = false)
```

Comment:
Should this maybe mirror an example in the BCL? Like `Type.GetType`:

https://docs.microsoft.com/en-us/dotnet/api/system.type.gettype?view=netcore-3.1#System_Type_GetType_System_String_System_Boolean_

They named this `throwOnError`, in this case we can default it to `true`.

This is also an ABI break to use an optional parameter. Can we leave the old method signature in, and just call the other method? (default pass in `true`)

Reply:
I actually had done that at first, and just changed it back, per your suggestion. So now the parameter is `throwIfDeleted` (since non delete errors always throw) and I updated the API to maintain backward ABI compatibility.